corehq/apps/app_manager/views/app_summary.py

```diff
@@ -456,7 +457,7 @@ def get_all_forms_row(self, module, form, language):
 PropertyRow = namedtuple('PropertyRow', CASE_SUMMARY_EXPORT_HEADER_NAMES)
 
 
-class DownloadCaseSummaryView(LoginAndDomainMixin, ApplicationViewMixin, View):
+class DownloadCaseSummaryView(LoginOrAPIKeyMixin, LoginAndDomainMixin, ApplicationViewMixin, View):
```

Comment:
Does having both of these mixins work as expected? Can you still download from HQ? 
I believe that `login_or_api_key` already encompasses the `login_and_domain` check, so you might only need one of these. 

I also think elsewhere we've used the `@method_decorator` pattern directly on the view. I'm conflicted about making another auth mixin here - my preference would be to just use the decorator directly.

Reply:
I'm the one who suggested that @dchukhin take the approach of adding a new mixin, but the decorator approach also sounds fine. Do you see any specific benefits to either approach? A decorator would be less code.

Reply:
I'm ok with either approach. The method decorator would also be slightly more specific (since it would target 1 request method, rather than all (currently 1) request methods). Let me know if you have preferences either way.

Reply:
I don't have a strong opinion either, but I do like the specificity of the decorator. It's also a bit more explicit when reading the code.

Reply:
Ok, it has been changed to a method decorator.